src/features/transactions/useHistoricalTransactions.ts

```diff
@@ -0,0 +1,171 @@
+import { useCustomCompareEffect } from "@react-hookz/web/esm";
+import { useWeb3React } from "@web3-react/core";
+
+import { Event } from "ethers";
+
+import { useAppDispatch } from "../../app/hooks";
+import Weth9 from "../../constants/Weth9";
+import {
+  checkPendingTransactionState,
+  getSwapArgsFromWrappedSwapForLog,
+  SwapEventArgs,
+} from "./transactionUtils";
+import {
+  setTransactions,
+  SubmittedTransactionWithOrder,
+} from "./transactionsSlice";
+import useSwapLogs from "./useSwapLogs";
+import useTransactionsFromLocalStorage from "./useTransactionsFromLocalStorage";
+
+const useHistoricalTransactions = () => {
+  const { chainId, library, account } = useWeb3React();
+  const { result: swapLogs, status: swapLogStatus } = useSwapLogs();
+  const {
+    transactions: localStorageTransactions,
+    setTransactions: setLocalStorageTransactions,
+  } = useTransactionsFromLocalStorage();
+  const dispatch = useAppDispatch();
+
+  useCustomCompareEffect(
+    () => {
+      if (swapLogStatus === "loading" || swapLogStatus === "not-executed")
+        return;
+      if (
+        !swapLogs ||
+        swapLogs.chainId !== chainId ||
+        swapLogs.account !== account
+      )
+        return;
+
+      const localTransactionsCopy = { all: [...localStorageTransactions.all] };
+
+      const updateStorage: () => void = async () => {
+        let lastLookSwapLogs: Event[] = [],
+          rfqSwapLogs: Event[] = [],
+          wrappedSwapLogs: Event[] = [];
+
+        if (!!swapLogs) {
+          lastLookSwapLogs = swapLogs.lastLookSwapLogs;
+          rfqSwapLogs = swapLogs.rfqSwapLogs;
+          wrappedSwapLogs = swapLogs.wrappedSwapLogs;
+        }
+
+        const reconcileLogs = async (
+          protocol: "last-look" | "request-for-quote",
+          logs: Event[],
+          isWrapped: boolean = false
+        ) => {
+          await Promise.all(
+            logs.map(async (swapLog) => {
+              const args = isWrapped
+                ? await getSwapArgsFromWrappedSwapForLog(swapLog)
```

Comment:
There's a `throw` in the code, but we should never hit it (every `WrappedSwap` event should be in a transaction with a `Swap` event). If not there has been a pretty major change so I took the view that "Crashing" the app when in dev mode (uncaught exception in promise) is actually good behaviour here. For production I think it will fail silently and historic transactions just won't load. I think this is ok?

Reply:
Sounds good to me 👍